model/DataStore/MetaDataDeliverySyncTask.php

```diff
@@ -0,0 +1,171 @@
+            $params['count']++;
+            try {
+                $this->getPersistDataService()->persist($params);
+                $report->setMessage('Success MetaData syncing for delivery: ' . $params['deliveryId']);
+            } catch (Throwable $exception) {
+                $this->logError(sprintf(
+                    'Failing MetaData syncing for delivery: %s with message: %s',
+                    $params['deliveryId'],
+                    $exception->getMessage()
+                ));
+
+                $report->setType(Report::TYPE_ERROR);
+                $report->setMessage('Failing MetaData syncing for delivery: ' . $params['deliveryId']);
+                $this->requeueTask($params);
+            }
+        }
+
+        return $report;
+    }
+
+    public function jsonSerialize()
+    {
+        return __CLASS__;
+    }
+
+    /**
+     * @throws InvalidServiceManagerException
+     */
+    private function getQueueDispatcher(): ConfigurableService
+    {
+        return $this->getServiceManager()->get(QueueDispatcher::SERVICE_ID);
+    }
+
+    /**
+     * @param $params
+     * @throws InvalidServiceManagerException
+     */
+    private function requeueTask($params): void
+    {
+        /** @var QueueDispatcher $queueDispatcher */
+        $queueDispatcher = $this->getQueueDispatcher();
+        $queueDispatcher->createTask(
+            $this,
+            $params,
+            __('Continue try to sync GCP of delivery "%s".', $params['deliveryId'])
+        );
+    }
+
+    private function getPersistDataService(): PersistDataService
+    {
+        return $this->getServiceLocator()->get(PersistDataService::class);
+    }
+
+    /**
+     * @throws common_exception_Error
+     * @throws core_kernel_persistence_Exception
+     */
+    private function prepareData($params)
+    {
+        $params['count'] = $params['count'] ?? 0;
+        if (!isset($params['deliveryMetaData'], $params['testMetaData'], $params['testUri'], $params['itemMetaData'])) {
+            $compiler = $this->getMetaDataCompiler();
+            //DeliveryMetaData
+            $deliveryResource = new core_kernel_classes_Resource($params['deliveryId']);
+            $params['deliveryMetaData'] = $compiler->compile($deliveryResource);
+            //test MetaData
+            $test = $this->getTest($deliveryResource);
+            $params['testUri'] = $this->getTestUri($deliveryResource);
+            $params['testMetaData'] = $compiler->compile($test);
+            //Item MetaData
+            $params['itemMetaData'] = $this->getItemMetaData($test, $compiler);
+        }
+
+        return $params;
+    }
+    private function getItemMetaData(core_kernel_classes_Resource $test, ResourceJsonMetadataCompiler $compiler): array
+    {
+        /** @var taoQtiTest_models_classes_QtiTestService $testService */
+        $testService = $this->getServiceLocator()->get(taoQtiTest_models_classes_QtiTestService::class);
```

Comment:
Can we extract it to separate method?

Reply:
this is the only place this is used , do you think it will improve the readability by extracting this ? 
i don't see any value added for this